src/raftstore/store/peer.rs

```diff
@@ -686,30 +712,104 @@ impl Peer {
             }
             vec![]
         } else {
-            self.handle_raft_committed_entries(ready.committed_entries.take().unwrap())
+            let committed_entries = ready.committed_entries.take().unwrap();
+            // leader needs to update lease.
+            let mut to_be_updated = self.is_leader();
+            if !to_be_updated {
+                // It's not leader anymore, we are safe to clear proposals. If it becomes leader
+                // again, the lease should be updated when election is finished, old proposals
+                // have no effect. Although this may allow old uuids to be reused before old
+                // callbacks are called.
+                self.proposals.clear();
+            }
+            for entry in committed_entries.iter().rev() {
+                // raft meta is very small, can be ignored.
+                self.raft_log_size_hint += entry.get_data().len() as u64;
+                to_be_updated = to_be_updated &&
```

Comment:
This is exactly what it's doing. If it update lease successfully, `maybe_update_lease` will not be called again.